pysidt/sidt.py

```diff
@@ -866,21 +907,36 @@ def estimate_uncertainty(self):
             self.node_uncertainties.update(
                 {node.name: 1.0 for i, node in enumerate(nodes)}
             )
+        
+        for node in self.nodes.values():
+            n = node.rule.num_data
+            if n > 1:
+                t = scipy.stats.t.ppf((1 + confidence_level) / 2, n - 1)
+                node.rule.uncertainty = t * self.node_uncertainties[node.name] / np.sqrt(n)
+        
+        for node in self.nodes.values():
+            n = node.rule.num_data
+            if n == 1:
+                node.rule.uncertainty = node.parent.rule.uncertainty
+
 
     def assign_depths(self):
         root = self.root
         _assign_depths(root)
 
-    def evaluate(self, mol):
+    def evaluate(self, mol, estimate_uncertainty=False):
         """
         Evaluate tree for a given possibly labeled mol
         """
-        out = 0.0
+        pred = 0.0
+        unc = 0.0
         decomp = self.decomposition(mol)
         for d in decomp:
             children = self.root.children
             node = self.root
-            out += node.rule
+            pred += node.rule.value
+            if estimate_uncertainty:
+                unc += node.rule.uncertainty ** 2
```

Comment:
I was recently working with our uncertainties and I'm pretty sure our node uncertainties are in variance not standard deviation so I don't think this should be squared.

Reply:
I added this line of code to calculate the uncertainty for rule, and didn't read it from the node_uncertainty dictionary:

```
                t = scipy.stats.t.ppf((1 + confidence_level) / 2, n - 1)
                node.rule = Rule(value=data_mean, uncertainty=t * np.std(data) / np.sqrt(n), num_data=n)
```

so the uncertainty under rule is the standard deviation. 

We should probably do the same for the node_uncertainty dict